### PR TITLE
Funnel Groups filtering

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

Allow filtering by groups in funnels

Goes in after https://github.com/PostHog/posthog/pull/6939

## How did you test this code?

Moaar tests

### CODE REVIEW

File: ee/clickhouse/models/group.py [hunk 1]
```diff
@@ -36,6 +36,6 @@ def get_aggregation_target_field(
     aggregation_group_type_index: Optional[int], event_table_alias: str, distinct_id_table_alias: str
 ) -> str:
     if aggregation_group_type_index is not None:
-        return f"$group_{aggregation_group_type_index}"
+        return f"{event_table_alias}.$group_{aggregation_group_type_index}"
```
Comment (macobo) [inline ee/clickhouse/models/group.py:39]: Good spot

File: ee/clickhouse/queries/funnels/test/test_funnel.py [hunk 2]
```diff
@@ -1463,3 +1463,195 @@ def test_funnel_aggregation_with_groups(self):
...
+    @snapshot_clickhouse_queries
+    def test_funnel_group_aggregation_with_groups_entity_filtering(self):
+        GroupTypeMapping.objects.create(team=self.team, group_type="organization", group_type_index=0)
+        GroupTypeMapping.objects.create(team=self.team, group_type="company", group_type_index=1)
+
+        create_group(team_id=self.team.pk, group_type_index=0, group_key="org:5", properties={"industry": "finance"})
+        create_group(team_id=self.team.pk, group_type_index=0, group_key="org:6", properties={"industry": "technology"})
+
+        create_group(team_id=self.team.pk, group_type_index=1, group_key="company:1", properties={})
+        create_group(team_id=self.team.pk, group_type_index=1, group_key="company:2", properties={})
```
Comment (macobo) [inline ee/clickhouse/queries/funnels/test/test_funnel.py:1476]: Extract method?

File: ee/clickhouse/queries/funnels/test/test_funnel.py [hunk 3]
```diff
@@ -1463,3 +1463,195 @@ def test_funnel_aggregation_with_groups(self):
...
+
+        filters = {
```
Comment (macobo) [inline ee/clickhouse/queries/funnels/test/test_funnel.py:1478]: I gave this feedback earlier as well:

I think the code structure here kind of smells.

We have:

```
setup_groups
filter_object
setup_event_data
run
assert
```

Put the filter data right next to the run instead of having it here? Instead of extracting the hash perhaps have

```
filter = Filter(data=..., team=self.team)
result = ClickhouseFunnel(filter, self.team).run()
```

This applies to other tests as well.

File: ee/clickhouse/queries/test/test_paths.py [hunk 4]
```diff
@@ -1866,7 +1866,7 @@ def test_path_removes_duplicates(self):
             ],
         )
 
-    @test_with_materialized_columns(["$current_url"])
+    @test_with_materialized_columns(["$current_url", "$screen_name"])
```
Comment (macobo) [inline ee/clickhouse/queries/test/test_paths.py:1869]: Q: What changed here? Seems unrelated.

**Review by macobo — APPROVED** [review]
Some testing-related nits but lgtm otherwise.

File: ee/clickhouse/queries/test/test_paths.py (hunk 4 above)
Comment (neilkakkar) [inline ee/clickhouse/queries/test/test_paths.py:1869]: random Paths failing test, I'm guessing used to pass earlier because no proper teardown earlier on materialized columns, and some new truncating on events table by @EDsCODE ?

File: ee/clickhouse/queries/funnels/test/test_funnel.py (hunk 3 above)
Comment (neilkakkar) [reply ee/clickhouse/queries/funnels/test/test_funnel.py:1478]: whoops, I moved the filter down, but not the data 🤦 . Thanks, will do

**Comment by neilkakkar** [comment]
Actually, will resolve the tests nit in the immediate next PR - have to do the same changes there as well.
